build.ps1

```diff
@@ -0,0 +1,77 @@
+[CmdletBinding(PositionalBinding = $false)]
+param(
+    [string] $VersionSuffix = "loc" + ((Get-Date).ToUniversalTime().ToString("yyyyMMddHHmm")),
+    [string] $ArtifactsPath = (Join-Path $PWD "artifacts"),
+    [string] $BuildConfiguration = "Release",
+
+    [bool] $RunBuild = $true,
+    [bool] $RunTests = $true
+)
+
+$ErrorActionPreference = "Stop"
+$Stopwatch = [System.Diagnostics.Stopwatch]::StartNew()
+
+function Task {
+    [CmdletBinding()] param (
+        [Parameter(Mandatory = $true)] [string] $name,
+        [Parameter(Mandatory = $false)] [bool] $runTask,
+        [Parameter(Mandatory = $false)] [scriptblock] $cmd
+    )
+
+    if ($cmd -eq $null) {
+        throw "Command is missing for task '$name'. Make sure the starting '{' is on the same line as the term 'Task'. E.g. 'Task `"$name`" `$Run$name {'"
+    }
+
+    if ($runTask -eq $true) {
+        Write-Host "`n------------------------- [$name] -------------------------`n" -ForegroundColor Cyan
+        $sw = [System.Diagnostics.Stopwatch]::StartNew()
+        & $cmd
+        Write-Host "`nTask '$name' finished in $($sw.Elapsed.TotalSeconds) sec."
+    }
+    else {
+        Write-Host "`n------------------ Skipping task '$name' ------------------" -ForegroundColor Yellow
+    }
+}
+
+
+Task "Init" $true {
+
+    if ($ArtifactsPath -eq $null) { "Property 'ArtifactsPath' may not be null." }
+    if ($BuildConfiguration -eq $null) { throw "Property 'BuildConfiguration' may not be null." }
+    if ((Get-Command "dotnet" -ErrorAction SilentlyContinue) -eq $null) { throw "'dotnet' command not found. Is .NET Core SDK installed?" }
+
+    Write-Host "VersionSuffix: $VersionSuffix"
+    Write-Host "ArtifactsPath: $ArtifactsPath"
+    Write-Host "BuildConfiguration: $BuildConfiguration"
+    Write-Host ".NET Core SDK: $(dotnet --version)`n"
+
+    Remove-Item -Path $ArtifactsPath -Recurse -Force -ErrorAction Ignore
+    New-Item $ArtifactsPath -ItemType Directory -ErrorAction Ignore | Out-Null
+    Write-Host "Created artifacts folder '$ArtifactsPath'"
+}
+
+Task "Build" $RunBuild {
+
+    dotnet msbuild "/t:Restore;Build;Pack" "/p:CI=true" `
```

Comment:
Not against using `dotnet msbuild`, but just wonder why you used that over using `dotnet build`?

Reply:
Because the dotnet CLI still is quite limited. It e.g. can't accept an empty VersionSuffix, so one would have to write a big `if ($VersionSuffix -eq $null) { dotnet build... } else { dotnet build --VersionSuffix $VersionSuffix }` with duplicate parameters. 

It also can't build & pack in one step which is a lot faster. With separate commands you again have to duplicate all the parameters etc.

Believe me, I've tried to exclusively use normal `dotnet` commands but I ran into so many issues (especially with older SDK versions) that this by far is the best way. This is also how it's done e.g. in [Dapper](https://github.com/StackExchange/Dapper/blob/07fe543f1530b33818c9ce989bf0672e145c954d/build.ps1#L123).

The regular `dotnet ***` commands [just call msbuild](https://github.com/dotnet/cli/blob/d0780d81e571abfcdcd1ed2ab85f0cc4a155528e/src/dotnet/commands/dotnet-build/BuildCommand.cs#L30) under the hood anyway.

Reply:
Gotcha. That makes sense.